Intermittent test failure: test_get_live_experiments_in_previous_namespaces

Seen locally.  These lists should be ordered so not sure what we missed here?

```py
======================================================================
FAIL: test_get_live_experiments_in_previous_namespaces (experimenter.experiments.tests.test_models.TestNimbusExperiment.test_get_live_experiments_in_previous_namespaces)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/unittest/case.py", line 57, in testPartExecutor
    yield
  File "/usr/local/lib/python3.11/unittest/case.py", line 623, in run
    self._callTestMethod(testMethod)
    ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/unittest/case.py", line 579, in _callTestMethod
    if method() is not None:
    ^^^^^^^^^^^^^^^^^
  File "/experimenter/experimenter/experiments/tests/test_models.py", line 2777, in test_get_live_experiments_in_previous_namespaces
    self.assertEqual(
  File "/usr/local/lib/python3.11/unittest/case.py", line 873, in assertEqual
    assertion_func(first, second, msg=msg)
    ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/unittest/case.py", line 1079, in assertListEqual
    self.assertSequenceEqual(list1, list2, msg, seq_type=list)
    ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/unittest/case.py", line 1061, in assertSequenceEqual
    self.fail(msg)
  File "/usr/local/lib/python3.11/unittest/case.py", line 703, in fail
    raise self.failureException(msg)
    ^^^^^^^^^^^^^^^^^
AssertionError: Lists differ: ['realigned-user-facing-parallelism', 're-eng[83 chars]ity'] != ['re-engineered-user-facing-artificial-intell[83 chars]ity']

First differing element 0:
'realigned-user-facing-parallelism'
're-engineered-user-facing-artificial-intelligence'

- ['realigned-user-facing-parallelism',
-  're-engineered-user-facing-artificial-intelligence',
? ^

+ ['re-engineered-user-facing-artificial-intelligence',
? ^

+  'realigned-user-facing-parallelism',
   'right-sized-upward-trending-flexibility']
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4360)

--- a/experimenter/experimenter/experiments/tests/test_models.py
+++ b/experimenter/experimenter/experiments/tests/test_models.py
@@ -2701,6 +2701,7 @@ class TestNimbusExperiment(TestCase):
         experiments = {
             slug: NimbusExperimentFactory.create_with_lifecycle(
                 NimbusExperimentFactory.Lifecycles.LIVE_APPROVE_APPROVE,
+                slug=slug,
                 application=NimbusExperiment.Application.DESKTOP,
                 channel=NimbusExperiment.Channel.RELEASE,
                 firefox_min_version=NimbusExperiment.Version.FIREFOX_129,
@@ -2714,7 +2715,6 @@ class TestNimbusExperiment(TestCase):
         experiment = NimbusExperimentFactory.create_with_lifecycle(
             NimbusExperimentFactory.Lifecycles.CREATED,
             application=NimbusExperiment.Application.DESKTOP,
-            slug="slug2",
             channel=NimbusExperiment.Channel.RELEASE,
             firefox_min_version=NimbusExperiment.Version.FIREFOX_129,
             firefox_max_version=NimbusExperiment.Version.FIREFOX_130,
@@ -2722,17 +2722,13 @@ class TestNimbusExperiment(TestCase):
             feature_configs=[feature],
         )
 
-        matching_experiments = experiment.live_experiments_in_namespace
-        self.assertEqual(len(matching_experiments), 3)
         self.assertEqual(
-            list(matching_experiments.values_list("slug", flat=True)),
-            sorted(
-                [
-                    experiments["bar"].slug,
-                    experiments["baz"].slug,
-                    experiments["foo"].slug,
-                ]
-            ),
+            list(experiment.live_experiments_in_namespace),
+            [
+                experiments["bar"].slug,
+                experiments["baz"].slug,
+                experiments["foo"].slug,
+            ],
         )
 
     def test_get_live_experiments_do_not_exist_in_previous_namespaces(self):

